src/pages/workspace/WorkspacesListPage.js

```diff
@@ -106,55 +111,155 @@ function dismissWorkspaceError(policyID, pendingAction) {
+    const [policyNameToDelete, setPolicyNameToDelete] = useState(null);
 
+    const confirmDeleteAndHideModal = () => {
+        Policy.deleteWorkspace(policyIDToDelete, [], policyNameToDelete);
+        setIsDeleteModalOpen(false);
+    };
     /**
      * Gets the menu item for each workspace
      *
      * @param {Object} item
-     * @param {Number} index
      * @returns {JSX}
      */
-    function getMenuItem(item, index) {
-        const keyTitle = item.translationKey ? translate(item.translationKey) : item.title;
-        const isPaymentItem = item.translationKey === 'common.wallet';
+    const getMenuItem = useCallback(
+        ({item, index}) => {
+            const keyTitle = item.translationKey ? translate(item.translationKey) : item.title;
+
+            const policyID = item.policyID;
+
+            const threeDotsMenuItems = [
+                {
+                    icon: Expensicons.Trashcan,
+                    text: translate('workspace.common.delete'),
+                    onSelected: () => {
+                        setPolicyIDToDelete(policyID);
+                        setPolicyNameToDelete(item.title);
+                        setIsDeleteModalOpen(true);
+                    },
+                },
+                {
+                    icon: Expensicons.Hashtag,
+                    text: translate('workspace.common.goToRoom', {roomName: CONST.REPORT.WORKSPACE_CHAT_ROOMS.ADMINS}),
+                    onSelected: () => Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(item.adminRoom)),
+                },
+                {
+                    icon: Expensicons.Hashtag,
+                    text: translate('workspace.common.goToRoom', {roomName: CONST.REPORT.WORKSPACE_CHAT_ROOMS.ANNOUNCE}),
+                    onSelected: () => Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(item.announceRoom)),
+                },
+            ];
+
+            return (
+                <OfflineWithFeedback
+                    key={`${keyTitle}_${index}`}
+                    pendingAction={item.pendingAction}
+                    errorRowStyles={styles.ph5}
+                    onClose={item.dismissError}
+                    errors={item.errors}
+                >
+                    <PressableWithoutFeedback
+                        accessibilityRole="button"
+                        style={[styles.mh5, styles.mb3]}
+                        onPress={() => {
+                            item.action();
+                        }}
+                    >
+                        <WorkspacesListRow
+                            title={keyTitle}
+                            menuItems={threeDotsMenuItems}
+                            workspaceIcon={item.icon}
+                            ownerAccountID={currentUserPersonalDetails.accountID}
+                            workspaceType={item.type}
+                            currentUserPersonalDetails={currentUserPersonalDetails}
```

Comment:
`WorkspacesListRow` already uses `withCurrentUserPersonalDetails` HOC. We should use the HOC either in the parent or child, not both.